assets/src/edit-story/utils/getFirstFrameOfVideo.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Returns an image of the first frame of a given video.
+ *
+ * @param {string} src Video src URL.
+ * @return {Promise<string>} The extracted image in base64-encoded format.
+ */
+function getFirstFrameOfVideo( src ) {
+	const video = document.createElement( 'video' );
+	video.muted = true;
+	video.crossOrigin = 'anonymous';
+	video.preload = 'metadata';
+	video.currentTime = 0.5; // Needed to seek forward.
+
+	return new Promise( ( resolve, reject ) => {
+		video.addEventListener( 'error', reject );
+
+		video.addEventListener( 'canplay', () => {
+			const canvas = document.createElement( 'canvas' );
+			canvas.width = video.videoWidth;
+			canvas.height = video.videoHeight;
+
+			const ctx = canvas.getContext( '2d' );
+			ctx.drawImage( video, 0, 0, canvas.width, canvas.height );
+
+			canvas.toBlob( resolve, 'image/jpeg' );
+		} );
+
+		video.src = src;
+	} );
+}
```

Comment:
This unfortunately doesn't test easily, as jest mocks the video object and never invokes the events even if loaded with a data uri. So, for this one I'd just say that we rely on this working elsewhere. Maybe link to where we stole it from in the first place.

Reply:
I wrote this for the old stories editor. Not sure how helpful a link to that one is 🤷‍♂

Reply:
Puppeteer tests could be possible, but not with MP4 videos.

Reply:
At the moment, we only support mp4 which is not supported in open source version of chrome, making e2e tests hard.